src/audio/pulseaudio/SDL_pulseaudio.c

```diff
@@ -752,27 +772,56 @@ SourceInfoCallback(pa_context *c, const pa_source_info *i, int is_last, void *da
             spec.callback = NULL;
             spec.userdata = NULL;
 
-            SDL_AddAudioDevice(SDL_TRUE, i->description, &spec, (void *) ((size_t) i->index+1));
+            if (add) {
+                SDL_AddAudioDevice(SDL_TRUE, i->description, &spec, (void *) ((size_t) i->index+1));
```

Comment:
This (and its likes) too need changing from size_t to (u)intptr_t I guess. (although this one seems to have been here for some time.)

Reply:
I'll do a sweep through pulseaudio.c in a separate PR; for now I've got intptr_t in all the new lines at least!